### DISCUSSION TITLE
U2F Metadata

### DISCUSSION THREAD

**Discussion by coelho** [zitadel/zitadel#4429] [Ideas]
Zitadel should collect U2F Metadata on registration (if not already) and display this to admins & users.
See: https://developers.yubico.com/U2F/Attestation_and_Metadata/JSON_Format.html

We are interested in this for:
1. Audits
2. Enforcing usage of only company assigned keys

It would also be useful to allow enforcing a whitelist of vendors, but that's not immediately required as we can just add do automated audits for now.

Thanks for the great work @ Zitadel Team 🥇

**Comment by fforootd** [comment]
Great idea, I think we do not store to many metadata right now.

But it would make for a great datasource to create interesting reports :grin: 

@livio-a what data is stored ATM?

**Reply by livio-a** [reply]
We do not store any metadata ATM. 
BTW we use WebAuthN for both passwordless and 2FA. For the later we do not request for user verification.
I fully support the idea to restrict to certain keys / certificates, but from what i know the metadata / webauthn extension implementation are different between the vendors. E.g. the linked metadata format is specific to Yubico.

**Reply by coelho** [reply]
@livio-a Sounds like the first step would be to start storing the raw metadata asap & then figure out parsing later. 
Would allow to grandfather old keys into full support once it is ready.
